scriptlets/check_for_snap_changes

```diff
@@ -37,20 +37,21 @@ all_complete() {
 
     # tail -n +2: remove the header
     # awk 'NF {print $2}': print the second column on non-empty lines (i.e. the status)
-    # grep -q -E "Doing|Undoing|Wait": succeed when changes are still ongoing
+    # grep -q -E "...": succeed when changes are still ongoing or pending
     cat $OUTPUT | \
     tail -n +2 | \
     awk 'NF {print $2}' | \
-    grep -q -E "Doing|Undoing|Wait"
+    grep -q -E "\b(Doing|Undoing|Wait|Do|Undo)\b"
 
     RESULT=$?
     if [ "$RESULT" -eq 0 ]; then
-        # the change is still ongoing: display output as a diagnostic and return 1
-        cat "$OUTPUT"
+        # changes are still ongoing or pending: display output as a diagnostic and return 1
+        cat "$OUTPUT" | grep -E "\b(Doing|Undoing|Wait|Do|Undo)\b"
         rm "$OUTPUT"
         return 1
     else
```

Comment:
Completely optional, but it looks like you can drop the else block completely. If RESULT was 0 then it already returned, so we can just drop out of the if and handle the other branch here without an "else"